### PR TITLE
Improve error message when there are different types

### PR DESCRIPTION
**Pull request by lucassouto**
When there are variables of different types, but with the same representation, the error message is confusing as the example below:

```
FAIL: test_update (v1.tests.tests_views.tests_patient_scheduling_views.PatientSchedulingViewTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/lucassouto/nilo/hippocrates-api/hippocrates_api/v1/tests/tests_views/tests_patient_scheduling_views.py", line 238, in test_update
    self.assertResponseUpdated(expected_item=expected, response=response)
  File "/home/lucassouto/.cache/pypoetry/virtualenvs/hippocrates-api-k4F-OvWg-py3.9/lib/python3.9/site-packages/drf_kit/tests.py", line 69, in assertResponseUpdated
    self.assertResponse(
  File "/home/lucassouto/.cache/pypoetry/virtualenvs/hippocrates-api-k4F-OvWg-py3.9/lib/python3.9/site-packages/drf_kit/tests.py", line 103, in assertResponse
    self.assertResponseMatch(expected=expected_body, received=body)
  File "/home/lucassouto/.cache/pypoetry/virtualenvs/hippocrates-api-k4F-OvWg-py3.9/lib/python3.9/site-packages/drf_kit/tests.py", line 194, in assertResponseMatch
    raise AssertionError(msg)
AssertionError: There's 1 fields that differ
- updated_at: Received `2021-12-29T11:48:21Z`, but expected `2021-12-29T11:48:21Z`
```

This PR adds the variable's type as a prefix when they are of different types.

### CODE REVIEW

**Review by elvisgs — APPROVED** [review]

**Review by elvisgs — APPROVED** [review]

**Review by lucasgomide — COMMENTED** [review]
By following [conventional commits](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type) I would use `refactor:`  prefix instead

**Review by joaodaher — CHANGES_REQUESTED** [review]
Do not commit those temp files generated by tests.

**Comment by lucassouto** [comment]
@joaodaher, @lucasgomide asked to insert the files, I will remove them then

**Comment by joaodaher** [comment]
It makes no sense adding files that tests will create every time.

File: drf_kit/tests.py [hunk 1]
```diff
@@ -184,6 +184,9 @@ def _compare(expected_item, received_item):
                 return {}
             except AssertionError:
                 msg = f"Received `{received_item}`, but expected `{expected_item}`"
+                if not isinstance(received_item, type(expected_item)):
+                    msg = f"Received `{type(received_item)} - {received_item}`, but expected `{type(expected_item)} - {expected_item}`"  # pylint: disable=C0301
```
Comment (joaodaher) [inline drf_kit/tests.py:188]: Split the line instead of suppressing the message.

File: test_app/tests/tests_test_helpers.py [hunk 2]
```diff
@@ -178,7 +179,7 @@ def test_match_embedded(self):
         self._assert_not_match(expected, received, message)
 
         received = {"name": "Harry", "age": 13}
-        message = "There's 1 fields that differ\n- age: Received `13`, but expected `{'amount': 13, 'min_age': 0}` "
+        message = "There's 1 fields that differ\n- age: Received `<class 'int'> - 13`, but expected `<class 'dict'> - {'amount': 13, 'min_age': 0}` "  # pylint: disable=C0301
```
Comment (joaodaher) [inline test_app/tests/tests_test_helpers.py:182]: Split the line instead of suppressing the message.

File: test_app/tests/tests_test_helpers.py [hunk 3]
```diff
@@ -272,3 +273,10 @@ def test_match_list(self):
             "- friends: Received `13`, but expected to `['Hermione', 'Hagrid', 'Dumbledore']` "
         )
         self._assert_not_match(expected, received, message)
+
+    def test_match_error_types(self):
+        expected = {"birthday": datetime(2022, 1, 1, 0, 0, 0)}
+        received = {"birthday": "2022-01-01 00:00:00"}
+
+        message = "There's 1 fields that differ\n- birthday: Received `<class 'str'> - 2022-01-01 00:00:00`, but expected `<class 'datetime.datetime'> - 2022-01-01 00:00:00` "  # pylint: disable=C0301
```
Comment (joaodaher) [inline test_app/tests/tests_test_helpers.py:281]: Split the line instead of suppressing the message.

**Review by joaodaher — APPROVED** [review]
Great addition!
Thank you a lot!

**Pull request merged**